src/services/forgot-password.ts

```diff
@@ -0,0 +1,31 @@
+import "server-only";
+
+import { randomInt, createHash } from "crypto";
+import { and, eq } from "drizzle-orm";
+
+import { FORGOT_PASSWORD_TOKEN_EXPIRY_TIME } from "@/lib/constants";
+import { db } from "@/lib/db";
+import { forgotPasswordTokens } from "@/lib/db/schema";
+
+/**
+ * creates and inserts a 6-digit verification token into the database
+ *
+ */
+export const insertForgotPasswordToken = async (email: string) => {
+  await db
+    .delete(forgotPasswordTokens)
+    .where(and(eq(forgotPasswordTokens.identifier, email)));
+
+  const token = randomInt(100000, 999999);
+
+  const { createHash } = require('crypto');
+  const hashedPassword = createHash('sha256').update(email + token.toString()).digest('hex');
```

Comment:
don't use require. and use top level imports (why are we importing this everytime the function is called?)

generating a code then hashing it is kinda redundant, 

jwts are very good for generating temporary tokens like this that without having to make a trip to the db (thus less latency), but up to you how you want to implement it